rosapi/src/rosapi/params.py

```diff
@@ -226,21 +234,21 @@ def _get_param_names(node_name):
     # This method is called in a service callback; calling a service of the same node
     # will cause a deadlock.
     global _parent_node_name
-    if node_name == _parent_node_name:
+    if node_name == _parent_node_name or node_name == _node.get_fully_qualified_name():
         return []
 
     client = _node.create_client(ListParameters, f"{node_name}/list_parameters")
 
-    ready = client.wait_for_service(timeout_sec=5.0)
+    ready = client.wait_for_service(timeout_sec=_timeout_sec)
     if not ready:
         raise RuntimeError("Wait for list_parameters service timed out")
```

Comment:
This function can crash if the `wait_for_service` call fails as the RuntimeError is not catched anywhere. The easiest way to make this call fail is to have a node which doesn't start it's parameter services. Here's an example:
```Python
import rclpy
rclpy.init();
node = rclpy.create_node('rosapi_test_node', start_parameter_services=False)
rclpy.spin(node)
```
As this is still related to the issue you are trying to fix (crashes during parameter retrieval), could you also fix that?

IMO we shouldn't wait 5 seconds for service to become available as this will greatly increase the response time when we are running nodes without parameter servers. If we assume the rosapi node has enough time to spin and gather information about the ROS graph, we should be good with just returning an empty list if the `list_parameters` service is not available right away:
```suggestion
    if not client.service_is_ready():
        return []
```

Reply:
Hope you don't mind me changing this myself

Reply:
hey @bjsowa! Thanks so much for finishing this up. I've had the flu and wasn't able to get anything done.